homeassistant/components/group/sensor.py

```diff
@@ -665,19 +675,40 @@ def _get_valid_units(self) -> set[str | None]:
 
         If device class is set and compatible unit of measurements.
         If device class is not set, use one unit of measurement.
+        Only calculate valid units if there are no valid units set.
         """
-        if (
-            device_class := self.device_class
-        ) in UNIT_CONVERTERS and self.native_unit_of_measurement:
+        # If we once have a set of valid units we should not recalculate it
+        # when all states needs to be considered
```

Comment:
I don't understand this comment

Reply:
I modified it and placed it correctly in the first if-statement